standard/patterns.md

```diff
@@ -157,13 +169,233 @@ var_pattern
+- Otherwise, if *type* is omitted, and the input value is of type `object`, `System.ITuple`, or some type that can be converted to `System.ITuple` by an implicit reference conversion, and no *identifier* appears among the subpatterns, then the match uses `System.ITuple`.
+- Otherwise, the pattern is a compile-time error.
+
+The order in which subpatterns are matched at runtime is unspecified, and a failed match might not attempt to match all subpatterns.
+
+> *Example*: Here, we deconstruct an expression result and match the resulting values against the corresponding nested patterns:
+>
+> <!-- Example: {template:"standalone-console-without-using", name:"PositionalPattern1", ignoredWarnings:["CS8321"]} -->
+> ```csharp
+> static string Classify(Point point) => point switch
+> {
+>     (0, 0) => "Origin",
+>     (1, 0) => "positive X basis end",
+>     (0, 1) => "positive Y basis end",
+>     _ => "Just a point",
+> };
+> 
+> public readonly struct Point
+> {
+>     public int X { get; }
+>     public int Y { get; }
+>     public Point(int x, int y) => (X, Y) = (x, y);
+>     public void Deconstruct(out int x, out int y) => (x, y) = (X, Y);
+> }
+> ```
+>
+> *end example*
+<!-- markdownlint-disable MD028 -->
+
+<!-- markdownlint-enable MD028 -->
+> *Example*: The names of tuple elements and Deconstruct parameters can be used in a positional pattern, as follows:
+>
+> <!-- Example: {template:"standalone-console", name:"PositionalPattern2", ignoredWarnings:["CS8321"], inferOutput:true} -->
+> ```csharp
+> var numbers = new List<int> { 10, 20, 30 };
+> if (SumAndCount(numbers) is (Sum: var sum, Count: var count))
+> {
+>     Console.WriteLine($"Sum of [{string.Join(" ", numbers)}] is {sum}");
+> }
+>
+> static (double Sum, int Count) SumAndCount(IEnumerable<int> numbers)
+> {
+>     int sum = 0;
+>     int count = 0;
+>     foreach (int number in numbers)
+>     {
+>         sum += number;
+>         count++;
+>     }
+>     return (sum, count);
+> }
+> ```
+>
+> The output produced is
+>
+> ```console
+> Sum of [10 20 30] is 60
+> ```
+>
+> *end example*
+
+### §property-pattern-new-clause Property pattern
+
+A *property_pattern* checks that the input value is not `null`, and recursively matches values extracted by the use of accessible properties or fields.
+
+```ANTLR
+property_pattern
+    : type? property_subpattern simple_designation?
```

Comment:
I believe that `type` here can't be a nullable reference type or nullable value type. I suspect we need to state that.